ledger/simulation/tracer.go

```diff
@@ -251,35 +247,29 @@ func (tracer *evalTracer) makeOpcodeTraceUnit(cx *logic.EvalContext) OpcodeTrace
 }
 
 func (tracer *evalTracer) BeforeOpcode(cx *logic.EvalContext) {
-	currentOpcodeUnit := tracer.makeOpcodeTraceUnit(cx)
-
-	// logic sig opcode part
-	if cx.RunMode() != logic.ModeApp {
-		// do nothing for LogicSig ops
-		if !tracer.result.ReturnTrace() {
-			return
+	if cx.RunMode() == logic.ModeApp {
```

Comment:
Changes to this function weren't necessary, but I thought this reorganization was a little simpler